packages/components/src/TCRUserDashboard/DashboardNewsroomStripeConnect.tsx

```diff
@@ -0,0 +1,201 @@
+          {this.renderBody()}
+        </StripeContainer>
+      </>
+    );
+  }
+
+  private renderBody(): JSX.Element {
+    if (!this.props.channelData.currentUserIsAdmin) {
+      return (
+        <>
+          {/*@TODO/tobek When we have a flow for updating newsroom channel admins, add it here*/}
+          <ErrorIcon width={16} height={16} /> You are not an admin for this Newsroom, so you cannot edit its Stripe
+          settings. Please contact an admin to be added.
+        </>
+      );
+    } else if (!this.state.connectingStripe && this.props.channelData.isStripeConnected) {
+      const justConnected = this.qsParams.code;
+      return (
+        <>
+          <p>
+            {justConnected ? "Connection successful! You are now" : "You are currently"} accepting credit cards through
+            your connected Stripe account.
+          </p>
+          <p>
+            <InvertedButton size={buttonSizes.SMALL} href="https://dashboard.stripe.com" target="_blank">
+              Go to Your Stripe Dashboard<NorthEastArrow color={colors.accent.CIVIL_BLUE} />
+            </InvertedButton>
+          </p>
+          {!justConnected && this.renderStripeConnectButton("Connect a different Stripe account")}
+        </>
+      );
+    } else if (this.state.connectStripeError) {
+      return (
+        <>
+          <p>
+            <ErrorIcon width={16} height={16} /> Sorry, there was an error connecting your Newsroom to your Stripe
+            account:{" "}
+            <code>{JSON.stringify(this.state.connectStripeError.message || this.state.connectStripeError)}</code>.
+            Please try again.
+          </p>
+          {this.renderStripeConnectButton()}
+        </>
+      );
+    } else if (this.state.connectingStripe || this.qsParams.code) {
+      return <StyledDashboardLoadingMessage>Connecting your Stripe account</StyledDashboardLoadingMessage>;
+    } else if (this.qsParams.error) {
+      return (
+        <>
+          <p>
+            <ErrorIcon width={16} height={16} /> Error connecting your Stripe account: {this.qsParams.error}:{" "}
+            {this.qsParams.error_description}. Please try again.
+          </p>
+          {this.renderStripeConnectButton()}
+        </>
+      );
+    } else {
+      return (
+        <>
+          <p>
+            Allows you to accept credit card payments for your Boosts. Stripe fees will apply. The Boost amounts sent by
+            supporters will go directly into your connected Stripe account. Civil will not keep or hold any of your
+            proceeds.
+          </p>
+          {this.renderStripeConnectButton()}
+        </>
+      );
+    }
+  }
+
+  private renderStripeConnectButton(linkText?: string): JSX.Element {
+    // @TODO/toby Our stripe client ID (which should be different on production) should be in some config, but we don't have any config for components package, would have to pass in from dapp via prop or add to civil context.
```

Comment:
@dankins or anyone else, how should we do this? We could define in https://github.com/joincivil/Civil/blob/master/packages/dapp/.env or we could pass in as env variable into react app the way we are doing for `SENTRY_DSN`.

Reply:
this likely needs to be passed in the same way as SENTRY_DSN, I can make updates to the `orchestration` repo to set this. We could probably tack the entire `config` object into CivilContext so you can grab it from stuff in `components`

Reply:
Updated using this approach (passing in config through context). Added the client id to orchestration in https://github.com/joincivil/orchestration/pull/95/ but put the staging id in dapp `.env` too so that it works locally.